Use better-html gem

EiNSTeiN- opened the pull request:
This rewrites the Deprecated Classes linter to use `better-html` which is a ERB-aware html parser that I wrote some time ago. I would like to use this parser in other linters too.

@jeremyhansonfinger @nwtn

EiNSTeiN- commented on lib/erb_lint/linters/deprecated_classes.rb:
```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
+require 'active_support/core_ext/module/delegation'
+require 'active_support/core_ext/class/attribute_accessors'
```
unfortunately better-html assumes these Rails extensions are present...

rafaelfranca replied:
In this case better_html should require them and also add `activesupport` as runtime dependency.

EiNSTeiN- replied:
That is indeed what will eventually happen, but that's not a reason to hold this PR

rafaelfranca replied:
True. Just remember that now `activesupport` is a runtime dependency of this gem so that needs to be changed in the gemspec. Also you need to add  `require 'active_support'` before those two lines since it is required to require the top level file before using any piece of the framework.

EiNSTeiN- replied:
I added `require 'active_support'` before these

jeremyhansonfinger commented:
@nwtn I don't really know enough about how this is used to confidently :+1:. You're definitely the better person to review.

EiNSTeiN- commented:
@nwtn ?

nwtn commented:
When I compare `dev erblint --lint-all` between this version and the current version there are differences. From a quick glance, those diffs don’t seem obviously wrong, but are they expected?

nwtn commented:
code lgtm, if the diffs are ok then :+1:

EiNSTeiN- commented:
I rebased to make sure differences are not caused by the nokogiri refactor that was reverted last week.

I had a look at differences and most of them are new instances of deprecated classes that are found by the html parser in `better-html` that were not found by the old regexp-based parser.

A few deprecated classes were not found by my parser (in `components/online_store/app/views/admin/theme_editor/_rte_templates.html.erb`) because they are located inside of `<script type="text/html">` elements; I added some more code to parse html out of script tags now.

rafaelfranca commented on erb_lint.gemspec:
```diff
@@ -12,6 +12,10 @@ Gem::Specification.new do |s|
+
   s.add_development_dependency 'rspec'
   s.add_development_dependency 'rubocop'
+  s.add_development_dependency 'activesupport', '~> 5.1'
```
This should be `s.add_dependency 'active_support'` and you don't need to specify the version since the features you are using are not 5.1 specific.

EiNSTeiN- commented:
I fixed dependencies in `0.0.5` so I updated this PR to reflect that
